src/Components/Browser.JS/src/Platform/Circuits/AutoReconnectCircuitHandler.ts

```diff
@@ -0,0 +1,48 @@
+import { CircuitHandler } from "./CircuitHandler";
+export class AutoReconnectCircuitHandler implements CircuitHandler {
+  modal: HTMLDivElement;
+  intervalHandle: number | null;
+
+  constructor(private maxRetries: number = 3, private retryInterval: number = 3000) {
+    const modal = document.createElement('div');
+    modal.className = "modal";
+    modal.appendChild(document.createTextNode("Attempting to reconnect to the server..."));
+    document.addEventListener("DOMContentLoaded", function (event) {
+      document.body.appendChild(modal);
+    });
+
+    this.modal = modal;
+    this.intervalHandle = null;
+  }
+  onConnectionUp() {
+    this.modal.style.display = 'none';
```

Comment:
Testing this is going to be tricky. With a local host, reconnect simulated using a forced disconnect is almost instant. I could try adding another handler that does busy waiting until the test has an opportunity to inspect things. Any suggestions on a better plan to test the UI?

Reply:
Since we have an internal-for-testing method `_internal.forceCloseConnection`, we could go further with that pattern and change it to a pair of methods like `_internal.goOffline` (disconnects, and sets a flag that makes `reconnect` into a no-op), and `_internal.goOnline` (clears the flag so that `reconnect` now does its normal thing).

Reply:
Unresolving just so you see this. Totally understand if you don't feel there's time to change this further before preview 3!

Reply:
Another possibility I've looked into, but don't have a definitive answer about, is whether Selenium can make the browser switch into "offline" mode then back "online" later. That would be ideal, but people on the internets seem disagreed about whether this is supported and what the code to do it would be.

Reply:
Would something like this work?
* Extend ComponentHub and add an additional InterruptConnection method.
* Have that method close the connection (as it's the same client proxy).
* Register that on the functional tests.
* Call that from your code to trigger it.